samples/sample_common/src/vaapi_utils_x11.cpp

```diff
@@ -51,114 +50,99 @@ X11LibVA::X11LibVA(void)
             m_display = m_x11lib.XOpenDisplay(VAAPI_X_DEFAULT_DISPLAY);
 
         if (NULL == m_display) sts = MFX_ERR_NOT_INITIALIZED;
+
         if (MFX_ERR_NONE == sts)
         {
             m_va_dpy = m_vax11lib.vaGetDisplay(m_display);
+
+            if (!m_va_dpy)
+            {
+                sts = MFX_ERR_NULL_PTR;
+            }
+        }
+#if !defined(X11_DRI3_SUPPORT)
+        if (MFX_ERR_NONE == sts)
+        {
             m_va_dpy_render = m_vax11lib.vaGetDisplay(m_display);
 
-            if (!m_va_dpy || !m_va_dpy_render)
+            if (!m_va_dpy_render)
             {
-                m_x11lib.XCloseDisplay(m_display);
                 sts = MFX_ERR_NULL_PTR;
             }
         }
         if (MFX_ERR_NONE == sts)
         {
             if (!setenv("LIBVA_DRIVER_NAME", PROCESSING_DRIVER_NAME, 1)) {
-            va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
-            sts = va_to_mfx_status(va_res);
-            if (MFX_ERR_NONE != sts) {
-                m_x11lib.XCloseDisplay(m_display);
-                }
+                va_res = m_libva.vaInitialize(m_va_dpy, &major_version, &minor_version);
+                sts = va_to_mfx_status(va_res);
                 unsetenv("LIBVA_DRIVER_NAME");
             } else {
                 sts = MFX_ERR_NOT_INITIALIZED;
```

Comment:
I treated lack of m_x11lib.XCloseDisplay(m_display) on that branch as a bug. Is it correct or there was some plan?